packages/core/src/components/Form.tsx

```diff
@@ -508,49 +520,70 @@ export default class Form<
 
     const mustValidate = !noValidate && liveValidate;
     let state: Partial<FormState<T, S, F>> = { formData, schema };
-    let newFormData = formData;
 
     if (omitExtraData === true && liveOmit === true) {
       const retrievedSchema = schemaUtils.retrieveSchema(schema, formData);
       const pathSchema = schemaUtils.toPathSchema(retrievedSchema, '', formData);
 
       const fieldNames = this.getFieldNames(pathSchema, formData);
 
-      newFormData = this.getUsedFormData(formData, fieldNames);
       state = {
-        formData: newFormData,
+        formData: this.getUsedFormData(formData, fieldNames),
       };
     }
 
-    if (mustValidate) {
-      const schemaValidation = this.validate(newFormData);
-      let errors = schemaValidation.errors;
-      let errorSchema = schemaValidation.errorSchema;
-      const schemaValidationErrors = errors;
-      const schemaValidationErrorSchema = errorSchema;
-      if (extraErrors) {
-        const merged = validationDataMerge(schemaValidation, extraErrors);
-        errorSchema = merged.errorSchema;
-        errors = merged.errors;
-      }
-      state = {
-        formData: newFormData,
-        errors,
-        errorSchema,
-        schemaValidationErrors,
-        schemaValidationErrorSchema,
-      };
-    } else if (!noValidate && newErrorSchema) {
+    if (!noValidate && newErrorSchema) {
       const errorSchema = extraErrors
         ? (mergeObjects(newErrorSchema, extraErrors, 'preventDuplicates') as ErrorSchema<T>)
         : newErrorSchema;
+
       state = {
-        formData: newFormData,
+        formData,
         errorSchema: errorSchema,
         errors: toErrorList(errorSchema),
       };
     }
-    this.setState(state as FormState<T, S, F>, () => onChange && onChange({ ...this.state, ...state }, id));
+
+    this.setState(state as FormState<T, S, F>, () => {
+      onChange && onChange({ ...this.state, ...state }, id);
+
+      if (mustValidate) {
+        // @ts-ignore
```

Comment:
What are Typescript error is being disabled?

Reply:
No idea, this is just to get it to compile, I would resolve all TS issues in the final PR